qcodes/dataset/sqlite_base.py

```diff
@@ -112,6 +114,47 @@ class ConnectionPlus(wrapt.ObjectProxy):
 SomeConnection = Union[sqlite3.Connection, ConnectionPlus]
 
 
+def upgrader(func: Callable[[SomeConnection], None]):
+    """
+    Decorator for database version upgrade functions. An upgrade function
+    must have the name `perform_db_upgrade_N_to_M` where N = M-1.
+
+    The decorator takes care of logging about the upgrade and managing the
+    database versioning
+    """
+    name_comps = func.__name__.split('_')
+    if not len(name_comps) == 6:
+        raise NameError('Decorated function not a valid upgrader. '
+                        'Must have name "perform_db_upgrade_N_to_M"')
```

Comment:
"... where M and N are numbers that are.... bla bla"

Reply:
I don't think this error message is the right place to explain how to use the upgrader. It's just there to catch a typo, not to teach users how to write an upgrade function (nor to give them a false sense of understanding of the upgrade process).